tracer/src/Datadog.Trace/Agent/IApiResponse.cs

```diff
@@ -82,5 +96,84 @@ public static bool ShouldRetry(this IApiResponse response)
 
             return shouldRetry;
         }
+
+        /// <summary>
+        /// Gets the <see cref="Encoding"/> represented by the charset defined in the content type header.
+        /// </summary>
+        /// <param name="contentTypeHeader">The raw content-type header, for example <c>"application/json;charset=utf-8"</c></param>
+        /// <returns>The encoding associated with the charset, or <see cref="EncodingHelpers.Utf8NoBom"/> if the content-type header was not provided,
+        /// if the charset was not provided, or if the charset was not recognized</returns>
+        public static Encoding GetCharsetEncoding(string contentTypeHeader)
+        {
+            // special casing application/json because it's so common
+            if (string.IsNullOrEmpty(contentTypeHeader)
```

Comment:
I don't know if we're trying to follow the HTTP spec, but according to [RFC9110, Section 8.3](https://www.rfc-editor.org/rfc/rfc9110#section-8.3), we probably shouldn't default to UTF-8:

> If a `Content-Type` header field is not present, the recipient MAY either assume a media type of `"application/octet-stream"` ([[RFC2046](https://www.rfc-editor.org/rfc/rfc9110#RFC2046)], [Section 4.5.1](https://www.rfc-editor.org/rfc/rfc2046#section-4.5.1)) or examine the data to determine its type.

Reply:
Irrelevant if we're assuming text (not binary) content. 😅

Reply:
> Irrelevant if we're assuming text (not binary) content. Which makes sense since this is the IApiResponse. Never mind. 😅

Yeah, we really need to make some bold assumptions unfortunately. The agent sends `text/plain` even when it's sending `application/json` 😅 

And as you say, certain mime types assume certain encodings. Realistically we'd probably be fine if we never checked this and only ever used UTF8 😅

Reply:
Cool. Just checking!